packages/EDA/src/stat-tools.ts

```diff
@@ -0,0 +1,261 @@
+    DG.Column.fromList(DG.COLUMN_TYPE.FLOAT, 'Sum of squares', [anovaTable.SSbn, anovaTable.SSwn, anovaTable.SStot]),
+    DG.Column.fromList(DG.COLUMN_TYPE.INT, 'Degrees of freedom', [anovaTable.DFbn, anovaTable.DFwn, anovaTable.DFtot]),
+    DG.Column.fromList(DG.COLUMN_TYPE.FLOAT, 'Mean square', [anovaTable.MSbn, anovaTable.MSwn, null]),
+    DG.Column.fromList(DG.COLUMN_TYPE.FLOAT, 'F-statistics', [anovaTable.Fstat, null, null]),
+    DG.Column.fromList(DG.COLUMN_TYPE.FLOAT, 'p-value', [anovaTable.pValue, null, null]),
+    DG.Column.fromList(DG.COLUMN_TYPE.FLOAT, `${alpha}-critical value`, [Fcritical, null, null]),
+  ]);
+} // getOneWayAnovaDataframe
+
+/** Check correctness of significance level. */
+function checkSignificanceLevel(alpha: number) {
+  if ((alpha <= 0) || (alpha >= 1))
+    throw new Error(ERROR_MSG.INCORRECT_SIGNIFICANCE_LEVEL);
+}
+
+/** Compute unbiased variance.*/
+export function getVariance(sampleData: SampleData): number {
+  // The applied formulas can be found in [4] (see p. 63)
+  const size = sampleData.size;
+
+  if (size <= 0)
+    throw new Error(ERROR_MSG.INCORRECT_SAMPLE_SIZE);
+
+  if (size === 1)
+    return 0;
+  
+  return (sampleData.sumOfSquares - (sampleData.sum) ** 2 / size) / (size - 1);
+} // getVariance
+
+/** Checks equality of variances of 2 samples. F-test is performed.*/
+export function areVariancesOfTwoSamplesEqual(xSampleData: SampleData, ySampleData: SampleData, alpha: number  = 0.05): boolean {
+  // The applied approach can be found in [3]
+  const xSampleVariance = getVariance(xSampleData);
+  const ySampleVariance = getVariance(ySampleData);
+
+  if (ySampleVariance === 0)
+    return (xSampleVariance === ySampleVariance);
+
+  const Fstat = xSampleVariance / ySampleVariance;
+  const Fcritical = jStat.centralF.inv(1 - alpha, xSampleData.size - 1, ySampleData.size - 1);
+
+  return (Fstat < Fcritical);
+} // areVariancesOfTwoSamplesEqual
+
+/** Checks equality of variances of factorized data. */
+export function areVariancesOfFactorizedDataEqual(factorizedData: Map<any, SampleData>): boolean {
+  const iter = factorizedData.keys();
+  let key = iter.next();
+  const xSampleData = factorizedData.get(key.value);
+
+  while (true) {
+    key = iter.next();
+
+    if (key.done)
+      break;
+
+    if(!areVariancesOfTwoSamplesEqual(xSampleData!, factorizedData.get(key.value)!))
+      return false;
+  }
+
+  return true;
+} // areVariancesOfFactorizedDataEqual
+
+/** Perform one-way analysis of variances. */
+export function oneWayAnova(
+  factorsCol: DG.Column, 
+  valuesCol: DG.Column,
+  alpha: number  = 0.05,
+  toCheckNormality: boolean = false, 
+  toCheckVariancesEqaulity: boolean = false): AnovaResults
+{
+  checkSignificanceLevel(alpha);
+
+  const values = ((valuesCol.type === DG.COLUMN_TYPE.FLOAT) || (valuesCol.type === DG.COLUMN_TYPE.INT)) 
+    ? valuesCol.getRawData() 
+    : valuesCol.toList();
+
+  const factors = factorsCol.toList();
```

Comment:
Why are we creating another copy of the list, can't we work with the column directly?

Reply:
For the case of **10M** rows, time of processing is:
- **312-497 ms** if we use a copy (const factors = factorsCol.toList() and then use factors[i]);
- **566-603 ms** if we work with a column directly.

What should be decreased in this case: time or spatial complexity?

Reply:
We should work with the column's getRawData, this is the best of both worlds, I think